morebits.js

```diff
@@ -1308,29 +1308,239 @@ Morebits.unbinder.getCallback = function UnbinderGetCallback(self) {
+			'mm': pad(m), 'm': m,
+			'ss': pad(s),
+			'dddd': udate.getDayName(), 'ddd': udate.getDayNameAbbrev(), 'd': udate.getDay(),
+			'DD': pad(D), 'D': D,
+			'MMMM': udate.getMonthName(), 'MMM': udate.getMonthNameAbbrev(), 'MM': pad(M), 'M': M,
+			'YYYY': Y, 'YY': Y % 100, 'Y': Y
+		};
+
+		var unbinder = new Morebits.unbinder(formatstr); // escape stuff between [...]
+		unbinder.unbind('\\[', '\\]');
+		unbinder.content = unbinder.content.replace(
+			/* Regex notes:
+			 * d(d{2,3})? matches exactly 1, 3 or 4 occurrences of 'd' ('dd' is treated as a double match of 'd')
+			 * Y{1,2}(Y{2})? matches exactly 1, 2 or 4 occurrences of 'Y'
+			 */
+			/H{1,2}|h{1,2}|m{1,2}|ss|d(d{2,3})?|D{1,2}|M{1,4}|Y{1,2}(Y{2})?|A/g,
+			function(match) {
+				return replacementMap[match];
+			}
+		);
+		return unbinder.rebind().replace(/\[(.*?)\]/g, '$1');
+	},
+
+	/**
+	 * Gives a readable relative time string such as "Yesterday at 6:43 PM" or "Last Thursday at 11:45 AM".
+	 * Similar to calendar in moment.js, but with time zone support.
+	 * @param {(string|number)} [zone=system] - 'system' (for browser-default time zone), 'utc' (for UTC),
+	 * or 'user' (for time zone according to user's site preferences, needs mediawiki.user dependency), or
+	 * specify a time zone as number of minutes past UTC
+	 * @returns {string}
+	 */
+	calendar: function(zone) {
+		var now = new Date(), thisdate = new Date(this.getTime()); // create a copy
+		now.setHours(0, 0, 0, 0); // zero out the hours, minutes, seconds and milliseconds - keeping only the date
+		thisdate.setHours(0, 0, 0, 0);
+		var datediff = (now.getTime() - thisdate.getTime()) / 86400000;
+		switch (true) {
+			case datediff === 0:
+				return this.format(Morebits.date.localeData.relativeTimes.thisDay, zone);
+			case datediff === 1:
+				return this.format(Morebits.date.localeData.relativeTimes.prevDay, zone);
+			case datediff > 0 && datediff < 7:
+				return this.format(Morebits.date.localeData.relativeTimes.lastWeek, zone);
+			case datediff === -1:
+				return this.format(Morebits.date.localeData.relativeTimes.nextDay, zone);
+			case datediff < 0 && datediff > -7:
+				return this.format(Morebits.date.localeData.relativeTimes.thisWeek, zone);
+			default:
+				return this.format(Morebits.date.localeData.relativeTimes.other, zone);
+		}
+	},
+
+	/**
+	 * Returns a regex that matches wikitext section titles such as ==December 2019== or
+	 * === Jan 2018 ===
+	 * @returns {RegExp}
+	 */
+	monthHeaderRegex: function() {
+		return new RegExp('^==+\\s*(?:' + this.getUTCMonthName() + '|' + this.getUTCMonthNameAbbrev() +
+			')\\s+' + this.getUTCFullYear() + '\\s*==+', 'mg');
+	},
+
+	/**
+	 * Creates a wikitext section header with the month and year.
+	 * @param {number} [level=2] - Header level (default 2)
+	 * @returns {string}
+	 */
+	monthHeader: function(level) {
+		level = level || 2;
+		var header = '';  // String.prototype.repeat is not supported in IE 11, so...
+		for (var i = 0; i < level; i++) {
```

Comment:
I didn't know of that one. Changed to that as its a one-liner. Besides, if I were to guess, I would say it'd be faster than the for loop, since the for loop is mostly javascript, while the former is mostly native C++ code, which obviously goes faster.